pdf/creator/table.go

```diff
@@ -215,34 +215,59 @@ func (table *Table) GeneratePageBlocks(ctx DrawContext) ([]*Block, DrawContext,
 
 		if cell.backgroundColor != nil {
 			// Draw background (fill)
-			rect := NewRectangle(ctx.X, ctx.Y, w, h)
+			border := NewBorder(ctx.X, ctx.Y, w, h, cell.borderLineStyle)
 			r := cell.backgroundColor.R()
 			g := cell.backgroundColor.G()
 			b := cell.backgroundColor.B()
-			rect.SetFillColor(ColorRGBFromArithmetic(r, g, b))
+			border.SetFillColor(ColorRGBFromArithmetic(r, g, b))
 			if cell.borderStyle != CellBorderStyleNone {
 				// and border.
-				rect.SetBorderWidth(cell.borderWidth)
-				r := cell.borderColor.R()
-				g := cell.borderColor.G()
-				b := cell.borderColor.B()
-				rect.SetBorderColor(ColorRGBFromArithmetic(r, g, b))
+				if cell.borderColorLeft != nil {
+					border.SetBorderColorLeft(ColorRGBFromArithmetic(cell.borderColorLeft.R(), cell.borderColorLeft.G(), cell.borderColorLeft.B()))
+				}
+				if cell.borderColorBottom != nil {
+					border.SetBorderColorBottom(ColorRGBFromArithmetic(cell.borderColorBottom.R(), cell.borderColorBottom.G(), cell.borderColorBottom.B()))
+				}
+				if cell.borderColorRight != nil {
+					border.SetBorderColorRight(ColorRGBFromArithmetic(cell.borderColorRight.R(), cell.borderColorRight.G(), cell.borderColorRight.B()))
+				}
+				if cell.borderColorTop != nil {
+					border.SetBorderColorTop(ColorRGBFromArithmetic(cell.borderColorTop.R(), cell.borderColorTop.G(), cell.borderColorTop.B()))
+				}
+				border.SetBorderWidthBottom(cell.borderWidthBottom)
+				border.SetBorderWidthLeft(cell.borderWidthLeft)
+				border.SetBorderWidthRight(cell.borderWidthRight)
+				border.SetBorderWidthTop(cell.borderWidthTop)
 			} else {
-				rect.SetBorderWidth(0)
+				border.SetBorderWidthBottom(0)
+				border.SetBorderWidthLeft(0)
+				border.SetBorderWidthRight(0)
+				border.SetBorderWidthTop(0)
 			}
-			err := block.Draw(rect)
+			err := block.Draw(border)
```

Comment:
Is it necessary to always draw the border, even if there is no border for a cell?
Makes the PDF output more cluttered many lines with 0 widths etc.
Preferable not to do, as long as it is easy to avoid.

Reply:
In previous implementation there is always `draw` call even if the width is 0. So I followed that to keep things similar.